pkg/api/types.go

```diff
@@ -138,6 +138,21 @@ type ObjectMeta struct {
 	// objects.  Annotation keys have the same formatting restrictions as Label keys. See the
 	// comments on Labels for details.
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// List of objects depended by this object. If ALL objects in the list have
+	// been deleted, this object will be garbage collected.
+	OwnerReferences []ObjectReference `json:"ownerReferences,omitempty"`
```

Comment:
@liggitt as proposed in #23656, deleting a parent object (e.g., a replicaset) may result in the deletion of a child object (e.g., a pod). I'd imagine If a user has the right to delete a replicaset, he should also has the right to delete a pod, is there any counterexample?

> Is cross namespace ownership allowed? I'd like to start with disallowing that…

We'll start with disallowing that. Do you mean I need to explicitly say this in the comment?